### DISCUSSION TITLE
"Windows Defender Application Control" compatibility

### DISCUSSION THREAD

**Discussion by ThiloLangbeinHUK** [PSAppDeployToolkit/PSAppDeployToolkit#853] [Ideas]
Maybe it's a good idea to make PSADT run with WDAC aka 'Application Control for Business' enabled.
<img width="528" alt="image" src="https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/assets/71653492/f41fbddc-33c0-4b43-9b6b-a5053248856f">
Code signing would be needed.
https://learn.microsoft.com/en-us/mem/intune/protect/endpoint-security-app-control-policy

**Comment by rathod-priyanka** [comment]
Windows Defender Application Control (WDAC) or Microsoft Device Guard, is a security feature introduced with Windows 10. This feature acts as a robust safeguard with advanced application control policies. It prevents any unauthorized code and ensures the integrity of the system. Only trusted applications can run, so the risk of malware and unauthorized software execution is minimal. Using a reliable [code signing certificate](https://shorturl.at/ikEx8) further enhances the trustworthiness of applications.

Here are key aspects and features of Windows Defender Application Control:

Code Integrity Policies
Controlled Folder Access
Windows Defender System Guard
Enterprise-Level Management
User Mode Code Integrity (UMCI)
Enforcement Modes
Windows Defender Application Control Flow Rules
Integration with Windows Defender Antivirus

**Comment by That-Annoying-Guy** [comment]
Project members are already aware and working on it.

The Solution involves replacing AppDeployToolkitMain.cs with AppDeployToolkitMain.dll !

**Reply by pvwc** [reply]
Is there any timeline on when the fix for this will be implemented

**Reply by mjr4077au** [reply]
Before the end of the year. All effort is going into v4.0 and there's been significant progress.
